CSharp_Net_module1_4_3_lab/CSharp_Net_module1_4_3_lab/Program.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections;
+
+namespace CSharp_Net_module1_4_3_lab
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            var animalsArray = new Animal[]
+            {
+                new Animal("Horse", 500),
+                new Animal("Cat", 3),
+                new Animal("Dog", 10),                
+                new Animal("Elephant", 3000),
+                new Animal("Wolf", 50)
+            };
+
+            var animals = new Animals(animalsArray);
+
+            ShowAnimals(animals);
+
+            foreach (var currentSortType in Enum.GetValues(typeof(SortTypes)))
+            {
+                var sortType = (SortTypes)currentSortType;
+
+                Console.WriteLine($"Sorting in {sortType}");
+                animals.Sort(sortType);
+                ShowAnimals(animals);
+            }
+        }
+
+        private static void ShowAnimals(Animals animals)
+        {
+            foreach (var currentAnimal in animals)
+            {
+                ShowAnimal((animal) => 
+                {
+                    Console.WriteLine($"{animal.Genus} - {animal.Weight}");
+                },
+                (Animal)currentAnimal);
```

Comment:
I don't think it's needed here

Reply:
Unfortunately explicit conversation is needed here, because I used non generic IEnumerator and it works with Object in foreach.
![image](https://user-images.githubusercontent.com/35422255/152683740-c6a428b0-e360-4519-9706-8725427be5e6.png)